src/utilities/path.ts

```diff
@@ -11,7 +11,7 @@ export type Path<
   P extends PathParams<T> = any
 > = {
   path: T,
-  params: P,
+  params: Identity<ExtractParamsFromPathString<T, P>>,
```

Comment:
Here's the implementation of path params. Essentially just moving the ExtractParamsFromPathString type here rather than using it in the route method types.